src/components/Terminal/Terminal.js

```diff
@@ -23,40 +31,79 @@ const Terminal = ({ address, modelName }) => {
       }
     };
     const creds = { macaroons };
+    const fitAddon = new FitAddon();
     const terminalInstance = setupTerminal(
       address,
       creds,
       modelName,
       terminalElement,
       switchFound
     );
+    terminalInstance.loadAddon(fitAddon);
+    fitAddonRef.current = fitAddon;
+
+    const resize = e => {
+      const viewPortHeight = window.innerHeight;
+      const mousePosition = e.clientY;
+      const newTerminalHeight =
+        viewPortHeight - mousePosition + terminalHeaderHeight;
+      if (newTerminalHeight >= minimumTerminalHeight) {
+        setTerminalHeight(newTerminalHeight);
+      }
+    };
+
+    const addResizeListener = e => {
+      if (e.target.classList.value === "p-terminal__header") {
+        document.addEventListener("mousemove", resize);
+      }
+    };
+
+    const removeResizeListener = () => {
+      document.removeEventListener("mousemove", resize);
+    };
+
+    // set resize listeners
+    window.addEventListener("mousedown", addResizeListener);
```

Comment:
Why are these events not registered using the react event system? That way it should handle attaching and detaching and simplify this event handling code.

```
<div className="p-terminal__header" onMouseUp={...} onMouseDown={...}>
```

Reply:
Issue raised: https://github.com/canonical-web-and-design/jaas-dashboard/issues/397